### PR TITLE
updated syntax to be compatible with julia 0.4.6

### PR DESCRIPTION
**Pull request by lruthotto**
and make all these warnings go away. Take a look if you want to pull this.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -50,4 +50,4 @@ provides(SimpleBuild,
              end)
           end), [libmumps_simple], os = :Unix)
 
-@BinDeps.install [:libmumps_simple => :libmumps_simple]
+@BinDeps.install Dict(:libmumps_simple => :libmumps_simple)
```
Comment (dpo) [inline deps/build.jl]: That's great, thanks. Would you mind using `@Compat` so we can keep supporting Julia 0.3?

Comment (lruthotto) [inline deps/build.jl]: Thanks for the hint. I can add that later. Shall I update the package requirement then, too? We should include https://github.com/JuliaLang/Compat.jl

Comment (dpo) [inline deps/build.jl]: Sure. Thanks!

Comment (lruthotto) [inline deps/build.jl]: I'm not sure we can be backward compatible with 0.3 since MPI.jl requires julia 0.4. Are you using MUMPS with juila 0.3?

Comment (dpo) [inline deps/build.jl]: Ah, well, that's a good reason to drop Julia 0.3 then!

**Comment by lruthotto** [comment]
no worries. I updated the requirement and squashed both commits. At least, I know about ```@compat``` now.

**Comment by dpo** [comment]
Thanks!

**Pull request merged**